apps/spruce/src/pages/adminSettings/Context.test.tsx

```diff
@@ -0,0 +1,76 @@
+import { act, renderHook, waitFor } from "@evg-ui/lib/test_utils";
+import { AdminSettingsTabRoutes } from "constants/routes";
+import { AdminSettingsProvider, useAdminSettingsContext } from "./Context";
+import { WritableAdminSettingsType } from "./tabs/types";
+
+describe("adminSettingsContext", () => {
+  it("ensure that tabs are initially saved", () => {
+    const { result } = renderHook(() => useAdminSettingsContext(), {
+      wrapper: AdminSettingsProvider,
+    });
+    expect(
+      result.current.getTab(AdminSettingsTabRoutes.Announcements).hasChanges,
+    ).toBe(false);
+  });
+  it("test check unsave settings", async () => {
+    const { result } = renderHook(() => useAdminSettingsContext(), {
+      wrapper: AdminSettingsProvider,
+    });
+
+    act(() => {
+      result.current.setInitialData({
+        [AdminSettingsTabRoutes.Announcements]: {
+          vars: [],
```

Comment:
nit. could we change this to be the real form state (`vars` doesn't actually exist in the form)